src/NLog/Internal/ExceptionHelper.cs

```diff
@@ -41,34 +42,87 @@ namespace NLog.Internal
     /// </summary>
     internal static class ExceptionHelper
     {
+        private const string LoggedKey = "NLog.ExceptionLoggedToInternalLogger";
+
+        /// <summary>
+        /// Mark this exception as logged to the <see cref="InternalLogger"/>.
+        /// </summary>
+        /// <param name="exception"></param>
+        /// <returns></returns>
+        public static void MarkedAsLoggedToInternalLogger(this Exception exception)
+        {
+            if (exception != null)
+            {
+                exception.Data[LoggedKey] = true;
+            }
+        }
+
+        /// <summary>
+        /// Is this exception logged to the <see cref="InternalLogger"/>? 
+        /// </summary>
+        /// <param name="exception"></param>
+        /// <returns><c>true</c>if the <paramref name="exception"/> has been logged to the <see cref="InternalLogger"/>.</returns>
+        public static bool IsLoggedToInternalLogger(this Exception exception)
+        {
+            if (exception != null)
+            {
+                return exception.Data[LoggedKey] as bool? ?? false;
+            }
+            return false;
+        }
+
+
         /// <summary>
-        /// Determines whether the exception must be rethrown.
+        /// Determines whether the exception must be rethrown and logs the error to the <see cref="InternalLogger"/> if <see cref="IsLoggedToInternalLogger"/> is <c>false</c>.
+        /// 
+        /// Advised to log first the error to the <see cref="InternalLogger"/> before calling this method.
         /// </summary>
-        /// <param name="exception">The exception.</param>
-        /// <returns>True if the exception must be rethrown, false otherwise.</returns>
+        /// <param name="exception">The exception to check.</param>
+        /// <returns><c>true</c>if the <paramref name="exception"/> must be rethrown, <c>false</c> otherwise.</returns>
         public static bool MustBeRethrown(this Exception exception)
         {
-            if (exception is StackOverflowException)
+            if (exception.MustBeRethrownImmediately())
             {
+                //no futher logging, because it can make servere exceptions only worse.
                 return true;
             }
 
-            if (exception is ThreadAbortException)
+            var isConfigError = exception is NLogConfigurationException
+                              || exception.GetType().IsSubclassOf(typeof(NLogConfigurationException));
```

Comment:
i have to double check this. I checked this earlier and then the conclusion was to keep it. Dunno why anymore. 

- [ ] check need for `IsSubclassOf`